### PR TITLE
Prevent PHP notices when LDAP does not provide "modifytimestamp" field

### PR DESCRIPTION
**Pull request by tomamplius**
Fix Notice error on LDAP sync with active directory because this error create 20MB of log in phplog file

| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | none

### CODE REVIEW

File: src/AuthLDAP.php [hunk 1]
```diff
@@ -1912,10 +1912,15 @@ public static function searchForUsers(
...
-                        );
+                        if ( isset($info[$ligne]['modifytimestamp']) ){
+                            $user_infos[$uid]["timestamp"] = self::ldapStamp2UnixStamp(
+                                $info[$ligne]['modifytimestamp'][0],
+                                $config_ldap->fields['time_offset']
+                            );
+			} else {
+                            $user_infos[$uid]["timestamp"] = 0;
+                        }
+
```
Comment (trasher) [inline src/AuthLDAP.php:1923]:
```suggestion
                        if (isset($info[$ligne]['modifytimestamp'])) {
                            $user_infos[$uid]["timestamp"] = self::ldapStamp2UnixStamp(
                                $info[$ligne]['modifytimestamp'][0],
                                $config_ldap->fields['time_offset']
                            );
			} else {
                            $user_infos[$uid]["timestamp"] = 0;
                        }

```

File: src/AuthLDAP.php [hunk 2]
```diff
@@ -1927,10 +1932,12 @@ public static function searchForUsers(
...
-                            $ldap_users[$uid] = self::ldapStamp2UnixStamp(
-                                $info[$ligne]['modifytimestamp'][0],
-                                $config_ldap->fields['time_offset']
-                            );
+                            if ( isset($info[$ligne]['modifytimestamp']) ){
+                                $ldap_users[$uid] = self::ldapStamp2UnixStamp(
+                                    $info[$ligne]['modifytimestamp'][0],
+                                    $config_ldap->fields['time_offset']
+                                );
+                            }
```
Comment (trasher) [inline src/AuthLDAP.php:1940]:
```suggestion
                            if (isset($info[$ligne]['modifytimestamp'])) {
                                $ldap_users[$uid] = self::ldapStamp2UnixStamp(
                                    $info[$ligne]['modifytimestamp'][0],
                                    $config_ldap->fields['time_offset']
                                );
                            }
```

File: src/AuthLDAP.php [hunk 3]
```diff
@@ -1912,10 +1912,15 @@ public static function searchForUsers(
...
-                        $user_infos[$uid]["timestamp"] = self::ldapStamp2UnixStamp(
-                            $info[$ligne]['modifytimestamp'][0],
-                            $config_ldap->fields['time_offset']
-                        );
+                        if (isset($info[$ligne]['modifytimestamp'])) {
+                            $user_infos[$uid]["timestamp"] = self::ldapStamp2UnixStamp(
+                                $info[$ligne]['modifytimestamp'][0],
+                                $config_ldap->fields['time_offset']
+                            );
+			} else {
```
Comment (cedric-anne) [inline src/AuthLDAP.php:1920]:
```suggestion
            } else {
```

File: src/AuthLDAP.php [hunk 4]
```diff
@@ -1912,10 +1912,15 @@ public static function searchForUsers(
...
-                        $user_infos[$uid]["timestamp"] = self::ldapStamp2UnixStamp(
-                            $info[$ligne]['modifytimestamp'][0],
-                            $config_ldap->fields['time_offset']
-                        );
+                        if (isset($info[$ligne]['modifytimestamp'])) {
+                            $user_infos[$uid]["timestamp"] = self::ldapStamp2UnixStamp(
+                                $info[$ligne]['modifytimestamp'][0],
+                                $config_ldap->fields['time_offset']
+                            );
+            } else {
```
Comment (cedric-anne) [inline src/AuthLDAP.php:1920]:
```suggestion
                            );
                        } else {
```

File: src/AuthLDAP.php [hunk 5]
```diff
@@ -1912,10 +1912,15 @@ public static function searchForUsers(
...
-                            $info[$ligne]['modifytimestamp'][0],
-                            $config_ldap->fields['time_offset']
-                        );
+                        if (isset($info[$ligne]['modifytimestamp'])) {
+                            $user_infos[$uid]["timestamp"] = self::ldapStamp2UnixStamp(
+                                $info[$ligne]['modifytimestamp'][0],
+                                $config_ldap->fields['time_offset']
+                            );
+                        } else {
+                            $user_infos[$uid]["timestamp"] = 0;
```
Comment (cedric-anne) [inline src/AuthLDAP.php:1921]:
```suggestion
                        } else {
                            $user_infos[$uid]["timestamp"] = '';
```

File: src/AuthLDAP.php [hunk 6]
```diff
@@ -1927,10 +1932,12 @@ public static function searchForUsers(
...
-                            $ldap_users[$uid] = self::ldapStamp2UnixStamp(
-                                $info[$ligne]['modifytimestamp'][0],
-                                $config_ldap->fields['time_offset']
-                            );
+                            if (isset($info[$ligne]['modifytimestamp'])) {
+                                $ldap_users[$uid] = self::ldapStamp2UnixStamp(
+                                    $info[$ligne]['modifytimestamp'][0],
+                                    $config_ldap->fields['time_offset']
+                                );
+                            }
```
Comment (cedric-anne) [inline src/AuthLDAP.php:1940]:
```suggestion
                            } else {
                                $ldap_users[$uid] = '';
                            }
```

**Review by cedric-anne — COMMENTED** [review]
This code is a real spaghetti dish, so it is best to ensure that the output has the same values as before.

**Review by cedric-anne — APPROVED** [review]
It will prevent notices when LDAP does not provide the "modifytimestamp" string, but maybe lack of this information induces side effects. However, this piece of code is really old, so I guess there is no major bug behind this case.

**Review by trasher — APPROVED** [review]

**Review by orthagh — APPROVED** [review]

**Pull request merged**